src/library/scala/collection/mutable/ListMap.scala

```diff
@@ -52,7 +52,8 @@ class ListMap[K, V]
   final override def clear(): Unit = { elems = List(); siz = 0 }
```

Comment:
Using List.apply instead of Nil or List.empty will be *slower* since it is dealing with varargs, right?
https://github.com/scala/scala/blob/2.13.x/src/library/scala/collection/Factory.scala#L94

Reply:
`List.apply()` is rewritten to `Nil` by the compiler. For all the other collections, you're right though.

```
$ scalac-ref heads/2.13.x <(echo 'class C { List() }') && jardiff C.class

+++ C.class.scalap
class C extends scala.AnyRef {
  def this() = { /* compiled code */ }
}

+++ C.class.asm
// class version 52.0 (52)
// access flags 0x21
public class C {


  // access flags 0x1
  public <init>()V
    ALOAD 0
    INVOKESPECIAL java/lang/Object.<init> ()V
    GETSTATIC scala/collection/immutable/Nil$.MODULE$ : Lscala/collection/immutable/Nil$;
    POP
    RETURN
    MAXSTACK = 1
    MAXLOCALS = 1
}
```

Reply:
@retronym Ow, that's … not so cool. Remove that rewrite? (Search-replace all List() to Nil / List.empty)

Reply:
It's a bit hard to search/replace is user code :) I don't think we can get rid of this one at the moment.

Reply:
add a ScalaFix for Scala 3 to remove this hack :)